packages/stack/src/views/Stack/CardStack.tsx

```diff
@@ -383,8 +434,31 @@ export default class CardStack extends React.Component<Props, State> {
     // For modals, usually we want the screen underneath to be visible, so also disable it there
     const isScreensEnabled = Platform.OS !== 'ios' && mode !== 'modal';
 
+    let floatingHeader;
+    if (headerMode === 'float') {
+      floatingHeader = renderHeader({
```

Comment:
I think this needs to be wrapped in a `React.Fragment` with `key` to preserve local state when being re-arranged.

Reply:
Good catch!